extensions/binding/org.eclipse.smarthome.binding.lirc/src/main/java/org/eclipse/smarthome/binding/lirc/internal/connector/LIRCConnector.java

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.lirc.internal.connector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.smarthome.binding.lirc.internal.config.LIRCBridgeConfiguration;
+import org.eclipse.smarthome.binding.lirc.internal.messages.LIRCButtonEvent;
+import org.eclipse.smarthome.binding.lirc.internal.messages.LIRCResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connector for communication with the LIRC server
+ *
+ * @author Andrew Nagle - Initial contributor
+ */
+public class LIRCConnector {
+
+    private static final Logger logger = LoggerFactory.getLogger(LIRCConnector.class);
+
+    private static List<LIRCEventListener> listeners = new ArrayList<LIRCEventListener>();
```

Comment:
This must not be static!
And: you might want to turn it into a Set, so you don't need to check whether an element is already in or not.